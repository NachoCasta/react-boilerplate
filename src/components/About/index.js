export { default } from "./AboutView";
export * from "./AboutReducer";
