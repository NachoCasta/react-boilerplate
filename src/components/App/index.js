export { default } from "./App";
export { default as AppNavBar } from "./NavBar";
export { default as AppContainer } from "./Container";
