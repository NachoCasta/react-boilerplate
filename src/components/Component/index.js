export { default } from "./ComponentView";
export * from "./ComponentReducer";
