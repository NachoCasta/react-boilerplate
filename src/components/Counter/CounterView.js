import React from "react";
import {
	Flex,
	Box,
	CardContainer,
	Heading,
	Button,
	RatingBadge
} from "components/UI";

const Counter = ({ onIncrement, onDecrement, counter }) => {
	return (
		<CardContainer width={[1, 3 / 4, 1 / 2, 1 / 3]}>
			<Heading textAlign="center">Counter</Heading>
			<Flex justifyContent="space-around">
				<Box width={[1]} p={1}>
					<Button width={1} onClick={onIncrement}>
						Increment
					</Button>
				</Box>
				<Box p={1}>
					<RatingBadge>{counter}</RatingBadge>
				</Box>
				<Box width={[1]} p={1}>
					<Button width={1} onClick={onDecrement}>
						Decrement
					</Button>
				</Box>
			</Flex>
		</CardContainer>
	);
};

export default Counter;
