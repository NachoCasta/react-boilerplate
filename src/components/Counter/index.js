export { default } from "./CounterContainer";
export * from "./CounterReducer";
