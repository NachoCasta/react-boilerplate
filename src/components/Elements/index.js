export { default } from "./ElementsContainer";
export * from "./ElementsReducer";
