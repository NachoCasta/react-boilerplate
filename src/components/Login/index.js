export { default } from "./LoginContainer";
export * from "./LoginReducer";
