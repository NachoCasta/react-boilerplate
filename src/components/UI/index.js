export * from "react-spinners";
export * from "./Springs";
export * from "./LinkList";
export * from "antd";
export {
    Container,
    Heading,
    Text,
    Link,
    RatingBadge,
    Hide
} from "pcln-design-system";
export { default as Box } from "./Box";
export { default as Flex } from "./Flex";
export { default as Card } from "./Card";
export { default as FadeCard } from "./FadeCard";
export { default as CardContainer } from "./CardContainer";
export { default as Form } from "./Form";
export { default as InputField } from "./InputField";
export { default as OnClickLink } from "./OnClickLink";
export { default as NavBar } from "./NavBar";
export { default as GradientBackgroundImage } from "./GradientBackgroundImage";
export { default as Input } from "./Input";
export { default as Select } from "./Select";
export { default as Button } from "./Button";
export { default as LoadingButton } from "./LoadingButton";
export { default as CircleButton } from "./CircleButton";
export { default as AddButton } from "./AddButton";
export { default as Scrollbar } from "./Scrollbar";
export { default as Code } from "./Code";
export { default as TextBlock } from "./TextBlock";
export { default as EditableTextBlock } from "./EditableTextBlock";
