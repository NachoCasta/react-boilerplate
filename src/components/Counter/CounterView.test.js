import React from "react";
import { shallow } from "enzyme";

import { RatingBadge } from "components/UI";
import CounterView from "./CounterView";

describe("CounterView", () => {
	let component;
	beforeEach(() => {
		component = shallow(<CounterView counter={5} />);
	});

	it("matches snapshot", () => {
		expect(component).toMatchSnapshot();
	});

	it("renders counter value", () => {
		expect(component.find(RatingBadge).text()).toBe("5");
	});
});
