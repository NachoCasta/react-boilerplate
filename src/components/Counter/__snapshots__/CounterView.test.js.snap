// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterView matches snapshot 1`] = `ShallowWrapper {}`;
